### PR TITLE
Add target name to portfolios map logic

### PR DESCRIPTION
**Pull request by cazapatamar**
Add target name to rendering layer logic in order to be able to send that info to backend

### CODE REVIEW

**Comment by cazapatamar** [comment]
@erikasv @ManuelStardust Ready!

File: src/utils/searchAPI.ts [hunk 1]
```diff
@@ -527,12 +527,13 @@ class SearchAPI {
...
     areaType: string,
     areaId: string,
+    targetName: string,
     portfolioId: number
   ): { request: Promise<Object>; source: CancelTokenSource } {
     const source = axios.CancelToken.source();
     return {
       request: SearchAPI.makeGetRequest(
-        `portfolios-ca/portfolios/layer?areaType=${areaType}&areaId=${areaId}&portfolioId=${portfolioId}`,
+        `portfolios-ca/portfolios/layer?areaType=${areaType}&areaId=${areaId}&targetName=${targetName}&portfolioId=${portfolioId}`,
```
Comment (erikasv) [inline src/utils/searchAPI.ts:536]: You didn't made any modification to the backend to do something with this param. And as long as I remember the portfolio layer is completely independent from the targets

File: src/pages/search/drawer/portfolios/conservationAreas/Targets.tsx [hunk 2]
```diff
@@ -86,7 +86,10 @@ class Targets extends React.Component<Props, State> {
                       );
                     if (portfoliosIds && portfoliosIds.size > 0) {
                       switchLayer(
-                        ["portfoliosCA", [...portfoliosIds].join("-")].join("-")
+                        [
+                          `portfoliosCA|${target.target_name}`,
+                          [...portfoliosIds].join(","),
+                        ].join("|")
```
Comment (erikasv) [inline src/pages/search/drawer/portfolios/conservationAreas/Targets.tsx:92]: This is too big and confusing to keep handling it with Arrays, I think it's better to construct the String directly with a template

File: src/utils/searchAPI.ts (hunk 1 above)
Comment (cazapatamar) [reply src/utils/searchAPI.ts]: You are right, i misunderstood, we have never have talk about relationship between targets and portfolios layer.

File: src/pages/search/drawer/portfolios/conservationAreas/Targets.tsx (hunk 2 above)
Comment (cazapatamar) [reply src/pages/search/drawer/portfolios/conservationAreas/Targets.tsx]: Done!

**Review by erikasv — APPROVED** [review]

**Pull request merged**